src/test/nexusTestUtils.js

```diff
@@ -52,104 +52,167 @@ fluid.test.nexus.assertContainsComponent = function (componentRoot, parentPath,
     jqUnit.assertValue(parentPath + " component contains " + childName, parent[childName]);
 };
 
+/**
+ * Assert that the object toTest contains a set of keys with given values.
+ * Can also be understood as "assert that the object expected is a subset of the object toTest".
+ * @param {Object} expected the expected keys and values.
+ * @param {Object} actual the object to test.
+ */
+fluid.test.nexus.assertKeyValues = function (expected, actual) {
+    for (var key in expected) {
+        var value = expected[key];
+        jqUnit.assertDeepEq(value, actual[key]);
```

Comment:
This must have only ever been working through some bizarre accident, e.g. actual[key] always being undefined, since you haven't put the required message first argument on - pls investigate!

Reply:
Well-spotted! Fixing this provokes a test failure that raises a question: since the readComponent handler JSON.stringifies its return value, the assertion here would need to parse it. The readDefaults handler does not stringify the object it returns. Is the implication that "raw JSON" is fine to send in an HTML response?

Reply:
Answered my own question pretty quickly. Turns out the response is stringified whether automatically as part of the response process, and I should parse it on the receiving end in any case :)

Reply:
Also, it shouldn't be an HTML response, the response type should be application/json https://stackoverflow.com/questions/477816/what-is-the-correct-json-content-type